pkg/storage/internalstorage/resource_storage.go

```diff
@@ -183,9 +183,6 @@ func (s *ResourceStorage) List(ctx context.Context, listObject runtime.Object, o
 
 	if amount != nil {
 		remain := *amount - offset - int64(len(objects))
-		if remain < 0 {
```

Comment:
remain can still be 0, right? e.g. `*amount` equals `offset+int64(len(objects))`.

Reply:
Yes, only when the current page has no data and the offset is too large, a negative number will be returned for front-end paging.

Reply:
nit: add a comment explaining `remain` might be negative and the reason why this is acceptable.(I think we used to add this logic for specific intension)

Reply:
> nit: add a comment explaining remain might be negative and the reason why this is acceptable.(I think we used to add this logic for specific intension)

That's right.
A negative remaining count is unconventional and some comments need to be added to clarify.